modules/javafx.controls/src/test/java/test/javafx/scene/control/TextInputControlTest.java

```diff
@@ -2043,6 +2048,105 @@ public void caretAndAnchorPositionAfterSettingText() {
         tk.firePulse();
     }
 
+    // Test for case 1 of JDK-8176270
+    @Test public void addingListenerWorks() {
+        VBox vBox = new VBox();
+        TextField textField = new TextField();
+        textField.setText("1234 5678");
+        vBox.getChildren().add(textField);
+        textField.selectedTextProperty()
+                 .addListener((observable -> {}));
+
+        Scene scene = new Scene(vBox);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    // Test for case 2 of JDK-8176270
+    @Test public void replaceSelectionWorks() throws Exception {
+        VBox vBox = new VBox();
+        TextField textField = new TextField();
+        textField.setText("1234 5678");
+        vBox.getChildren().add(textField);
+        textField.selectedTextProperty()
+                 .addListener((observable -> {}));
+
+        Scene scene = new Scene(vBox);
+        Stage stage = new Stage();
+        stage.setScene(scene);
+        stage.show();
+
+        textField.selectedTextProperty()
+                 .addListener(observable -> {
+                     // accessing the selectedTextProperty causes a
+                     // StringOutOfBoundsException
+                     observable.toString();
+                 });
+        textField.positionCaret(5);
+        Semaphore semaphore = new Semaphore(0);
+        Platform.runLater(semaphore::release);
```

Comment:
Since controls tests run using the StubToolkit, `Platform.runLater` does not do what you expect. If you need to run a pulse to get some code to run that normally would run as part of pulse, you can call that method directly. See other controls tests for how this is done.